File: update_dashboard.py
import json

with open('observability/grafana/provisioning/dashboards/medipass-dashboard.json', 'r', encoding='utf-8') as f:
    dashboard = json.load(f)

for panel in dashboard.get('panels', []):
    if 'targets' in panel:
        # Check if ms-apigateway is already there to avoid duplicates
        has_gateway = any('ms-apigateway' in t.get('expr', '') for t in panel['targets'])
        
        if not has_gateway:
            # We want to clone the first target but change the job and legend
            if len(panel['targets']) > 0:
                import copy
                new_target = copy.deepcopy(panel['targets'][0])
                
                # Replace job and legend
                new_target['expr'] = new_target['expr'].replace('ms-agendahub', 'ms-apigateway')
                new_target['legendFormat'] = new_target['legendFormat'].replace('AgendaHub', 'ApiGateway')
                new_target['refId'] = chr(ord(panel['targets'][-1]['refId']) + 1) if len(panel['targets'][-1]['refId']) == 1 else 'D'
                
                # Special cases for business metrics (they don't use job label usually)
                if 'medipass_' not in new_target['expr']:
                    panel['targets'].append(new_target)
        
        # Change rate to increase for 4xx and 5xx errors
        if panel.get('title', '').startswith('Tasa de Errores'):
            panel['title'] = panel['title'].replace('Tasa', 'Conteo')
            for t in panel['targets']:
                if 'rate(' in t['expr']:
                    t['expr'] = t['expr'].replace('rate(', 'increase(').replace('[1m]', '[5m]')
                    
        # Also let's change legend to not say 'Tasa' if it's count
        if panel.get('title', '').startswith('Conteo de Errores'):
            if panel['fieldConfig']['defaults'].get('unit') == 'reqps':
                panel['fieldConfig']['defaults']['unit'] = 'short'

with open('observability/grafana/provisioning/dashboards/medipass-dashboard.json', 'w', encoding='utf-8') as f:
    json.dump(dashboard, f, indent=2, ensure_ascii=False)
print('Dashboard updated successfully.')

File: update_dashboard.js
const fs = require('fs');
let data = JSON.parse(fs.readFileSync('observability/grafana/provisioning/dashboards/medipass-dashboard.json', 'utf8'));

data.panels.forEach(panel => {
    if (panel.targets) {
        let hasGw = panel.targets.some(t => t.expr && t.expr.includes('ms-apigateway'));
        if (!hasGw && panel.targets.length > 0) {
            let t0 = JSON.parse(JSON.stringify(panel.targets[0]));
            if (t0.expr && !t0.expr.includes('medipass_')) {
                t0.expr = t0.expr.replace('ms-agendahub', 'ms-apigateway');
                if (t0.legendFormat) {
                    t0.legendFormat = t0.legendFormat.replace('AgendaHub', 'ApiGateway');
                }
                t0.refId = String.fromCharCode(panel.targets[panel.targets.length-1].refId.charCodeAt(0) + 1);
                panel.targets.push(t0);
            }
        }
        
        if (panel.title && panel.title.includes('Tasa de Errores')) {
            panel.title = panel.title.replace('Tasa', 'Conteo');
            if (panel.fieldConfig && panel.fieldConfig.defaults) {
                panel.fieldConfig.defaults.unit = 'short';
            }
            panel.targets.forEach(t => {
                if (t.expr && t.expr.includes('rate(')) {
                    t.expr = t.expr.replace('rate(', 'increase(').replace('[1m]', '[5m]');
                }
            });
        }
    }
});

fs.writeFileSync('observability/grafana/provisioning/dashboards/medipass-dashboard.json', JSON.stringify(data, null, 2));
console.log('Done');
